### PR TITLE
Allow code submission

### PR DESCRIPTION
**Pull request by victorbalan**

### CODE REVIEW

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/SubmissionRepository.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package uno.cod.platform.server.core.repository;
+
+import org.springframework.data.jpa.repository.JpaRepository;
+import uno.cod.platform.server.core.domain.Submission;
+
+public interface SubmissionRepository extends JpaRepository<Submission, Long>{
```
Comment (Exordian) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/repository/SubmissionRepository.java]: add `@Repository`

File: platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/SubmissionService.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+import uno.cod.platform.server.core.domain.Task;
+import uno.cod.platform.server.core.repository.ResultRepository;
+import uno.cod.platform.server.core.repository.SubmissionRepository;
+import uno.cod.platform.server.core.repository.TaskRepository;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+@Service
```
Comment (Exordian) [inline platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/SubmissionService.java]: add `@Transactional`

File: platform-server-rest/src/main/java/uno/cod/platform/server/rest/RestUrls.java [hunk 3]
```diff
@@ -20,6 +20,8 @@
     public static final String TASKS = "/tasks";
     public static final String TASKS_ID = "/tasks/{id}";
 
+    public static final String RESULTS_TASKS_SUBMISSIONS = "/results/{resultId}/tasks/{taskId}/submissions";
```
Comment (Exordian) [inline platform-server-rest/src/main/java/uno/cod/platform/server/rest/RestUrls.java]: why do we submit a submission to a result ? doesn't the result derive from the submission?

Comment (flowlo) [inline platform-server-rest/src/main/java/uno/cod/platform/server/rest/RestUrls.java]: That's fine. A result may be computed from all (or only selected) Submissions. A Result links a User to a Challenge, and we relate the uploads (read: Submissions) to it.

**Comment by flowlo** [comment]
When I initially discussed this with @Exordian, I think we thought about handling uploaded submissions on the runtime side, storing them centralized, and giving away only normalized scores to the platform. Why has this changed?

**Comment by victorbalan** [comment]
i did not know that

**Pull request merged**